### PR TITLE
Create Arduino Library

### PR DESCRIPTION
**Pull request by headHUB**
Thank you for this software!  It works perfectly.  I am new to writing software so I wanted to contribute.  I took the repository and made it into an Arduino library so it can be downloaded as a zip file and installed through Arduino menu.  I also moved the sample arduino file so that it can ba accessed from the file menu.  I am in the middle of writing two more examples and will upload when done.

### CODE REVIEW

**Comment by headHUB** [comment]
hello?

**Comment by victorhurdugaci** [comment]
> hello?

Sorry, I was on vacation :) I'm back now

File: README.MD [hunk 1]
```diff
@@ -36,4 +30,4 @@ SerialProtocol is inspired by [Arduino-EasyTransfer](https://github.com/madsci10
 #### License
 
 Copyright [Victor Hurdugaci](http://victorhurdugaci.com). All rights reserved.  
-Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
\ No newline at end of file
+Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
```
Comment (victorhurdugaci) [inline README.MD:43]: Can you please add a new section at the end of this file called "Contributors" and add your name to it?

File: keywords.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#######################################
+# Syntax Coloring Map ArduinoSerialProtocol
+#######################################
+
+#######################################
+# Datatypes (KEYWORD1)
+#######################################
+
```
Comment (victorhurdugaci) [inline keywords.txt]: nitpick: there's an empty line here after `####` but no in the other sections

File: README.MD [hunk 3]
```diff
@@ -14,13 +14,7 @@ Total message size (bytes): Payload + 3
...
 #### Install Arduino Libraries
 
-Copy the `src\SerialProtocol` folder to:
-
-- Windows: `%USERPROFILE%\Victor\Documents\Arduino`
-- Linux: `~/sketchbook/libraries` or `/usr/share/arduino/libraries`
-- Mac: `Documents/Arduino/libraries`
-
-Note: You must restart the Arduino IDE if it was running before you copied the files.
+Download the Repository, Go to Sketch>Include Library>Add .zip Library> Select the Downloaded .zip File.
```
Comment (victorhurdugaci) [inline README.MD]: can you please make this a bit more explicit by saying "Download the repository as zip"? Someone might get confused by it and just git clone it.

Also nitpick on this line. `>` follow by spaces in some cases but not in others (consistency)

File: README.MD [hunk 4]
```diff
@@ -14,13 +14,7 @@ Total message size (bytes): Payload + 3
...
 
 #### Samples
```
Comment (victorhurdugaci) [inline README.MD:22]: I don't think the `samples` folder exist anymore in your change, right? Can you also update this part of the readme?

File: extras/raspberry_app/makefile [hunk 5]
```diff
@@ -11,6 +11,4 @@ clean:
 	rm $(TARGET)
 
 $(TARGET): 
-	g++ -I$(SERIAL_PROTOCOL_DIR) $(CPP_SOURCES) -o $(TARGET)
-	
-
+g++ -I$(SERIAL_PROTOCOL_DIR) $(CPP_SOURCES) -o $(TARGET)
```
Comment (victorhurdugaci) [inline extras/raspberry_app/makefile]: Why did you remove the indentation?

**Review by victorhurdugaci — CHANGES_REQUESTED** [review]
Hey @headHUB congratulations on your first contribution! 👏 

I left a few comments but overall the change looks good. Let me know when you addressed them and I take another look

File: README.MD (hunk 1 above)
Comment (headHUB) [reply README.MD:43]: I have completed this.  A github user per1234 contributed to my branch so should I include them as a contributor here?  I have no contact information except http://github.com/per1234.

File: README.MD (hunk 4 above)
Comment (headHUB) [reply README.MD:22]: I rewrote this whole section to be more user friendly.

**Comment by headHUB** [comment]
Now that I have made changes and committed to my branch, should I create a new request?  I've never done this before.  This is the first chance I have been able to help with this sort of thing.

File: README.MD (hunk 1 above)
Comment (headHUB) [reply README.MD:43]: I also suggest renaming repository to ArduinoSerialProtocol.  When you do I can adjust the readme to match

Comment (victorhurdugaci) [reply README.MD:43]: I'll leave the repository name like this for now

**Review by victorhurdugaci — APPROVED** [review]

**Comment by victorhurdugaci** [comment]
Congratulations on merging your first PR :)

**Comment by headHUB** [comment]
Thank you.  Ive borrowed so much code from so many places, I'm glad to be able to give back a little.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**